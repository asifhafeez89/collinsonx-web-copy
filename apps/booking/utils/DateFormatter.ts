import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';
import { DATE_TIME_FORMAT } from 'config/Constants';
import 'dayjs/locale/es';
dayjs.extend(utc);
dayjs.extend(timezone);

export function formatDate(
  dateval: Date,
  formattype: string,
  locale?: string
): string {
  dayjs.locale(locale ? locale : 'en');
  return dayjs(dateval).format(formattype);
}

export function formatTimezone(localTimeHour: string, timezone: string) {
  return dayjs.tz(localTimeHour, timezone).format(DATE_TIME_FORMAT);
}
