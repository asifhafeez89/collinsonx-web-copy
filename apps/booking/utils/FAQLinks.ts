export const FAQLink = (accountProvider?: string, locale?: string) => {
  const loungeKey_link_faq = `https://portal.loungekey.com/${locale}/elo/faq/`;
  const priorityPass_link_faq = `https://memberhelp.prioritypass.com/${locale}/support/home`;
  return accountProvider === 'LOUNGE_KEY'
    ? loungeKey_link_faq
    : accountProvider === 'PRIORITY_PASS'
    ? priorityPass_link_faq
    : '';
};
