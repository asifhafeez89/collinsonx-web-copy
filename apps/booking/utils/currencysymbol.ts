export const currencyMap: Record<string, string> = {
  GBP: String.fromCharCode(163),
};

export const getCurrencySymbol = (currency: string) =>
  currencyMap[currency] || currency;
