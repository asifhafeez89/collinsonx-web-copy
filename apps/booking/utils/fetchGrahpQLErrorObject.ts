// @ts-ignore
function fetchGrahpQLErrorObject(response, key: string) {
  if (!response && typeof response !== 'object') {
    return null;
  }

  if (!response[key]) {
    return null;
  }

  const error = response[key];

  if (error && 'graphQLErrors' in error) {
    const graphQLErrors = error.graphQLErrors;

    if (graphQLErrors && typeof graphQLErrors === 'object') {
      const data = error.graphQLErrors[0];
      if ('extensions' in data) {
        return data.extensions;
      }
    }
  }

  return null;
}

export default fetchGrahpQLErrorObject;
