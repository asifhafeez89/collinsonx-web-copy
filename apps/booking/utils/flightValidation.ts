const FLIGHT_VALIDATION = /([0-9]{0,1}[a-zA-Z]{1,3})([0-9]*)/gi;

export function validateFlightNumber(
  flight: string
): [boolean, string | undefined, string | undefined] {
  const match = [...flight.matchAll(FLIGHT_VALIDATION)];

  if (!match) {
    return [false, undefined, undefined];
  }

  const carrierLen = match[0]?.[1]?.length ?? 0;
  if (carrierLen < 1 || carrierLen > 3) {
    return [false, undefined, undefined];
  }

  const flightNoLen = match[0]?.[2]?.length ?? 0;
  if (flightNoLen < 1 || flightNoLen > 5) {
    return [false, undefined, undefined];
  }

  return [true, match[0][1], match[0][2]];
}
