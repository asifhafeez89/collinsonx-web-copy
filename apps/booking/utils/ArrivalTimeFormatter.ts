import { formatDate } from './DateFormatter';
import { TIME_FORMAT } from 'config/Constants';

export const arrivalTimeFormatter = (
  bookedFrom: string | undefined,
  lastArrival: string | undefined
) =>
  `${formatDate(new Date(`${bookedFrom}`), TIME_FORMAT)}
  -
  ${formatDate(new Date(`${lastArrival}`), TIME_FORMAT)}`;
