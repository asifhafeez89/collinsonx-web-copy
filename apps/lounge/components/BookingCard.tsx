import React from 'react';
import styled from '@collinsonx/design-system/styled';
import dayjs from 'dayjs';
import {
  Box,
  Button,
  Stack,
  Title,
  Text,
  Group,
} from '@collinsonx/design-system/core';
import BookingBadge from './BookingBadge';
import { MapPin } from '@collinsonx/design-system/assets/icons';
import NextImage from 'next/image';
import { BookingStatus } from '@collinsonx/utils';

// import generated types in the following way:
// import { Booking} from '@collinsonx/utils/generatedTypes/graphql';

const CardWrapper = styled.div`
  width: 343px;
  transition: 0.3s;
  border-radius: 5px;
  border: 1px solid #e9ecef;
  padding: 1rem;
  margin-bottom: 1rem;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.1), 0px 1px 3px rgba(0, 0, 0, 0.05);
`;

const ContentWrapper = styled.div`
  border-radius: 5px 5px 0 0;
`;

export interface BookingCardProps {
  id: string;
  name: string;
  location: string;
  imgUrl: string;
  status: BookingStatus;
  date: string;
  onClick: (bookingId: string) => void;
}

export default function BookingCard({
  id,
  name,
  location,
  imgUrl,
  status,
  date,
  onClick,
}: BookingCardProps) {
  return (
    <CardWrapper>
      <Stack spacing={16}>
        <Box
          w={309}
          h={189}
          sx={{
            position: 'relative',
            borderRadius: 4,
          }}
        >
          <NextImage fill src={imgUrl} alt={name} />
          <Box sx={{ position: 'absolute', right: 8, top: 8 }}>
            <BookingBadge status={status}>
              Booking {status.toLowerCase()}
            </BookingBadge>
          </Box>
        </Box>

        <ContentWrapper>
          <Stack spacing={8}>
            <Title fw={600} size={18}>
              {name}
            </Title>

            <Group spacing={8}>
              <MapPin />
              <Text>{location}</Text>
            </Group>

            <Title fw={600} size={18}>
              Date
            </Title>

            <Text>{dayjs(date).format('DD/MM/YYYY')}</Text>
          </Stack>
        </ContentWrapper>

        <Button variant="outline" fullWidth={true} onClick={() => onClick(id)}>
          View details
        </Button>
      </Stack>
    </CardWrapper>
  );
}
