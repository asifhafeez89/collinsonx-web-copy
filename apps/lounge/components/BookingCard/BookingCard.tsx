import React from 'react';
import styled from '@collinsonx/design-system/styled';
import dayjs from 'dayjs';
import {
  Box,
  Stack,
  Title,
  Text,
  Anchor,
} from '@collinsonx/design-system/core';
import BookingBadge from '../BookingBadge';
import NextImage from 'next/image';
import { BookingStatus } from '@collinsonx/utils';
import utc from 'dayjs/plugin/utc';
import {
  Warning as WarningIcon,
  Declined as DeclinedIcon,
} from '@collinsonx/design-system/assets/icons';
dayjs.extend(utc);

// import generated types in the following way:
// import { Booking} from '@collinsonx/utils/generatedTypes/graphql';

interface CardWrapperProps {
  nextVisit: boolean;
  firstArray: boolean;
}

const CardWrapper = styled.div<CardWrapperProps>`
  display: flex;
  flex-direction: ${(props) => (props.nextVisit ? 'column' : 'row')};
  flex-wrap: nowrap;
  width: 343px;
  transition: 0.3s;
  margin: 0;
  background-color: #fff;
  border-top: ${(props) =>
    props.firstArray && !props.nextVisit ? undefined : '2px solid #c8c9ca'};
  border-radius: ${(props) => (props.nextVisit ? '4px' : '0')};
  border: ${(props) => (props.nextVisit ? '2px solid #c8c9ca' : undefined)};
  padding: ${(props) => (props.nextVisit ? '1rem' : '1rem 0')};
`;

const CardImage = styled.img<CardWrapperProps>`
  width: ${(props) => (props.nextVisit ? '100%' : '107px !important')};
  height: ${(props) => (props.nextVisit ? '100%' : '107px !important')};
  margin-right: ${(props) => (props.nextVisit ? undefined : '24px')};
  float: left;
`;

const ContentWrapper = styled.div<CardWrapperProps>`
  border-radius: 5px 5px 0 0;
`;

const ContentStack = styled.div<CardWrapperProps>`
  display: flex;
  flex-direction: column;
  .card-title {
    margin-top: ${(props) => (props.nextVisit ? '8px' : '0')};
    font-size: ${(props) => (props.nextVisit ? '24px' : '16px')};
    line-height: ${(props) => (props.nextVisit ? '1.429em' : '1em')};
  }
`;
export interface BookingCardProps {
  id: string;
  name: string;
  location: string;
  imgUrl: string;
  status: BookingStatus;
  date: string;
  bookedFrom: string;
  nextVisit?: boolean;
  firstArray?: boolean;
  onClick: (bookingId: string) => void;
}

export interface BookingBadgeProps {
  fullBadge?: boolean;
}

export default function BookingCard({
  id,
  name,
  location,
  imgUrl,
  status,
  date,
  nextVisit = false,
  firstArray = false,
  onClick,
}: BookingCardProps) {
  return (
    <>
      <CardWrapper
        nextVisit={nextVisit}
        firstArray={firstArray}
        onClick={() => onClick(id)}
      >
        <Box
          sx={{
            position: 'relative',
          }}
        >
          <CardImage
            nextVisit={nextVisit}
            firstArray={firstArray}
            src={imgUrl}
            alt={name}
          />
          {!nextVisit && (
            <Box
              style={{
                position: 'absolute',
                top: 2,
                right: 24,
                outline: '8px solid #fff',
                borderBottomLeftRadius: 4,
              }}
            >
              <BookingBadge fullBadge={false} status={status}></BookingBadge>
            </Box>
          )}
        </Box>
        <Box>
          <Stack spacing={0}>
            <ContentWrapper nextVisit={nextVisit} firstArray={firstArray}>
              <ContentStack nextVisit={nextVisit} firstArray={firstArray}>
                <Title className={'card-title'} fw={600}>
                  {name}
                </Title>
                <Stack spacing={4}>
                  <Text mt={8} size={16}>
                    {location}
                  </Text>
                  <Text size={16}>{dayjs.utc(date).format('D MMMM YYYY')}</Text>
                  <Text size={16}>
                    {dayjs.utc(date).format('HH:mm')} lounge arrival time
                  </Text>
                  {nextVisit && (
                    <BookingBadge
                      fullBadge={true}
                      status={status}
                      mt={12}
                      h={24}
                    >
                      {status.toLowerCase()}
                    </BookingBadge>
                  )}
                </Stack>
              </ContentStack>
            </ContentWrapper>
          </Stack>
        </Box>
      </CardWrapper>
      {status === 'DECLINED' && (
        <Stack
          spacing={0}
          style={{
            paddingBottom: '24px',
          }}
        >
          <Box
            style={{
              display: 'flex',
              backgroundColor: '#FCD8D8',
              marginTop: '8px',
              padding: '8px 16px',
              border: '1px solid #F03E3E',
              borderRadius: '4px',
            }}
          >
            <DeclinedIcon
              style={{
                height: '28px',
                width: '28px',
                marginRight: '10px',
                color: '#C92A2A',
              }}
            />
            <Stack spacing={0}>
              <Text style={{ fontSize: '14px' }}>
                Unfortunately the lounge has declined your request.
                <Anchor
                  href="/"
                  target="_blank"
                  style={{
                    fontSize: '14px',
                    fontWeight: 600,
                    color: '#C92A2A',
                  }}
                >
                  See your messages
                </Anchor>
              </Text>
            </Stack>
          </Box>
        </Stack>
      )}
      {status === 'CANCELLED' && (
        <Stack
          spacing={0}
          style={{
            paddingBottom: '24px',
          }}
        >
          <Box
            style={{
              display: 'flex',
              backgroundColor: '#FCD8D8',
              marginTop: '8px',
              padding: '8px 16px',
              border: '1px solid #F03E3E',
              borderRadius: '4px',
            }}
          >
            <DeclinedIcon
              style={{
                height: '28px',
                width: '28px',
                marginRight: '10px',
                color: '#C92A2A',
              }}
            />
            <Stack spacing={0}>
              <Text style={{ fontSize: '14px' }}>
                Your cancellation has been confirmed.
                <Anchor
                  href="/"
                  target="_blank"
                  style={{
                    fontSize: '14px',
                    fontWeight: 600,
                    color: '#C92A2A',
                  }}
                >
                  See your messages
                </Anchor>
              </Text>
            </Stack>
          </Box>
        </Stack>
      )}
    </>
  );
}
