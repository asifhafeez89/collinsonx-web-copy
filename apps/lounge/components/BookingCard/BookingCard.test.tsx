import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import dayjs from 'dayjs';
import '@testing-library/jest-dom';
import BookingCard from './BookingCard';
import { BookingStatus } from '@collinsonx/utils';

const props = {
  id: '1',
  name: 'Booking Test',
  location: 'London',
  imgUrl:
    'https://preview.redd.it/w3kr4m2fi3111.png?auto=webp&v=enabled&s=670db506318b4c312b529c49c50e8a8c8423be21',
  status: BookingStatus.Confirmed,
  date: '2023-05-09T10:00:00.000Z',
  bookedFrom: '2023-05-09T10:00:00.000Z',
  nextVisit: true,
  firstArray: true,

  onClick: jest.fn(),
};

describe('BookingCard component test', () => {
  it('renders correctly', () => {
    const { container } = render(<BookingCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls onClick when card is clicked', () => {
    render(<BookingCard {...props} />);
    const testElement = screen.getByText('View booking');
    fireEvent.click(testElement);
    expect(props.onClick).toHaveBeenCalledWith(props.id);
  });

  it('checks if card information is displayed', () => {
    render(<BookingCard {...props} />);
    expect(screen.getByText(props.name)).toBeInTheDocument();
    expect(screen.getByText(props.location)).toBeInTheDocument();
    expect(screen.getByText('9 May 2023')).toBeInTheDocument();
    expect(screen.getByText('10:00 lounge arrival time')).toBeInTheDocument();
  });

  it('checks for correct time formatting', () => {
    render(<BookingCard {...props} date="2023-05-09T14:00:00.000Z" />);
    expect(screen.getByText(props.name)).toBeInTheDocument();
    expect(screen.getByText(props.location)).toBeInTheDocument();
    expect(screen.getByText('9 May 2023')).toBeInTheDocument();
    expect(screen.getByText('14:00 lounge arrival time')).toBeInTheDocument();
  });
});
