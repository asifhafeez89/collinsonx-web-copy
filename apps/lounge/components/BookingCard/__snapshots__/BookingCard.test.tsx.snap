// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingCard component test renders correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 343px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  margin: 0;
  background-color: #fff;
  border-top: 2px solid #c8c9ca;
  border-radius: 4px;
  border: 2px solid #c8c9ca;
  padding: 1rem;
}

.emotion-1 {
  position: relative;
}

.emotion-2 {
  width: 100%;
  height: 100%;
  float: left;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 0rem;
}

.emotion-6 {
  border-radius: 5px 5px 0 0;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-7 .card-title {
  margin-top: 8px;
  font-size: 24px;
  line-height: 1.429em;
}

.emotion-10 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  font-weight: 700;
  font-size: 2.125rem;
  line-height: 1.3;
  margin: 0;
  font-weight: 600;
}

.emotion-10:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-10:focus:not(:focus-visible) {
  outline: none;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 0.25rem;
}

.emotion-14 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: 1rem;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-top: 0.5rem;
}

.emotion-14:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-14:focus:not(:focus-visible) {
  outline: none;
}

.emotion-16 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: 1rem;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-16:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-16:focus:not(:focus-visible) {
  outline: none;
}

.emotion-20 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  font-size: 0.6875rem;
  height: 1.25rem;
  line-height: calc(1.25rem - 0.125rem);
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 calc(1rem / 1.5);
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: auto;
  text-transform: uppercase;
  border-radius: 2rem;
  font-weight: 700;
  letter-spacing: 0.015625rem;
  cursor: inherit;
  text-overflow: ellipsis;
  overflow: hidden;
  background: rgba(231, 245, 255, 1);
  color: #228be6;
  border: 0.0625rem solid transparent;
  border-radius: 4px;
  text-transform: none;
  font-size: 14px;
  line-height: 24;
  font-weight: 400;
  color: #25262B;
  padding: 12px 8px;
  background-color: #E3F1D0;
  border: 2px solid #74B816;
  overflow: visible;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  margin-top: 0.75rem;
  height: 1.5rem;
}

.emotion-20:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-20:focus:not(:focus-visible) {
  outline: none;
}

.emotion-21 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-24 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 1rem;
}

.emotion-24>* {
  box-sizing: border-box;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

<div
  class="emotion-0"
  data-testid="booking-card-wrapper"
>
  <div
    class="emotion-1"
  >
    <img
      alt="Booking Test"
      class="emotion-2"
      src="https://preview.redd.it/w3kr4m2fi3111.png?auto=webp&v=enabled&s=670db506318b4c312b529c49c50e8a8c8423be21"
    />
  </div>
  <div
    class="emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <div
        class="emotion-6"
      >
        <div
          class="emotion-7"
        >
          <h1
            class="emotion-8 emotion-9 card-title emotion-10"
          >
            Booking Test
          </h1>
          <div
            class="emotion-4 emotion-12"
          >
            <div
              class="emotion-8 emotion-14"
            >
              London
            </div>
            <div
              class="emotion-8 emotion-16"
            >
              9 May 2023
            </div>
            <div
              class="emotion-8 emotion-16"
            >
              10:00
               lounge arrival time
            </div>
            <div
              class="emotion-19 emotion-20"
            >
              <span
                class="emotion-21 emotion-22"
              >
                <div
                  class="emotion-23 emotion-24"
                >
                  Confirmed
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
